suffuse/root.go

```diff
@@ -0,0 +1,86 @@
+package suffuse
+
+import (
+  "os"
+  "bazil.org/fuse"
+)
+
+/** An interface encapsulating the essential fuse operations.
+ *  Each method returns nil if inapplicable. The fuse operations
+ *  use the first non-nil return value found for a given Path.
+ */
+type Rule interface {
+  MetaData(Path)*fuse.Attr
+  FileData(Path)[]byte
+  DirData(Path)[]fuse.Dirent
+  LinkData(Path)*Path
+}
+
+/** Eventually we'll read this kind of information out of a config
+ *  file, but for now it's hardcoded here.
+ */
+var Rules = []Rule {
+  IdRule{},  // The identity fs - passes underlying paths through unchanged
+  SedRule{}, // The sed fs - parses a '#...' suffix and applies it to the prefix path
+}
+
+/** Default implementations. A Rule struct can embed BaseRule
+ *  and then not have to implement irrelevant methods. There's
+ *  no default for Metadata because a filesystem which never
+ *  returns metadata is a useless filesystem.
+ */
```

Comment:
We should add something like:

> If a rule does not match, it should return `nil`. This is especially important in case of the methods that return an array.